server/src/main/java/org/opensearch/search/aggregations/MultiBucketConsumerService.java

```diff
@@ -153,10 +157,24 @@ public void accept(int value) {
                     );
                 }
             }
-            // check parent circuit breaker every 1024 calls
-            callCount++;
-            if ((callCount & 0x3FF) == 0) {
-                breaker.addEstimateBytesAndMaybeBreak(0, "allocated_buckets");
+            callCount.increment();
+            // tripping the circuit breaker for other threads in case of concurrent search
+            // if the circuit breaker has tripped for one of the threads already, more info
+            // can be found on: https://github.com/opensearch-project/OpenSearch/issues/7785
+            if (circuitBreakerTripped) {
+                throw new CircuitBreakingException(
+                    "Circuit breaker has tripped for one of the other threads",
+                    CircuitBreaker.Durability.PERMANENT
```

Comment:
Please change `CircuiteBreaker.Durability.PERMANENT` to `breaker.getDurability()`